File: tomgo/tests/testdata/equiv/leaf/scenario.go
// Equivalence scenario for Leaf.gom (built-in String slot).
package main

import (
	"fmt"

	"tomgen/leaf"
)

func main() {
	t1 := leaf.MakeLeaf()
	t2 := leaf.MakeLabel("hello")
	t3 := leaf.MakeLabel("hello")
	t4 := leaf.MakeLabel("world")

	fmt.Printf("t1=%v\n", t1)
	fmt.Printf("t2=%v\n", t2)
	fmt.Printf("shared-label=%v\n", t2 == t3)
	fmt.Printf("diff-labels=%v\n", t2 != t4)

	// Accessor: extract the String slot from a Label and print it back.
	lab := t2.(*leaf.LabelLeaf)
	fmt.Printf("label-s=%v\n", lab.S)
}

File: tomgo/tests/testdata/equiv/leaf/Scenario.java
// Equivalence scenario for Leaf.gom — Java reference side.
import leaf.types.Leaf;
import leaf.types.leaf.Label;

public class Scenario {
    public static void main(String[] args) {
        Leaf t1 = leaf.types.leaf.Leaf.make();
        Leaf t2 = Label.make("hello");
        Leaf t3 = Label.make("hello");
        Leaf t4 = Label.make("world");

        System.out.println("t1=" + t1);
        System.out.println("t2=" + t2);
        System.out.println("shared-label=" + (t2 == t3));
        System.out.println("diff-labels=" + (t2 != t4));

        Label lab = (Label) t2;
        System.out.println("label-s=" + lab.gets());
    }
}
